p2p/host/basic/basic_host_test.go

```diff
@@ -199,6 +199,12 @@ func TestHostProtoPreference(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// force the lazy negotiation to complete
+	_, err = s.Write(nil)
```

Comment:
This is now necessary because, unlike before, we now wait for identify to complete. This shouldn't slow anything down as we'd have to wait one round trip either way. Really, it should be _faster_ because identify will wait exactly one round-trip while a protocol negotiation could take multiple.

Reply:
what are the semantics of writing `nil` on streams?

Reply:
IIUC, nothing gets written over the wire, this just triggers everything down to that point.

Reply:
this relies on the specific behaviour of the msmux lazy stream to not nop on zero length writes